util/dvsim/utils.py

```diff
@@ -206,17 +206,43 @@ def md_results_to_html(title, css_path, md_text):
 
 
 def htmc_color_pc_cells(text):
-    '''This function finds cells in a html table that contains a "%" sign. It then
-    uses the number in front if the % sign to color the cell based on the value
-    from a shade from red to green. These color styles are encoded in ./style.css
-    which is assumed to be accessible by the final webpage.
+    '''This function finds cells in a html table that contain numerical values
+    (and a few known strings) followed by a single space and an identifier.
+    Depending on the identifier, it shades the cell in a specific way. A set of
+    12 color palettes for setting those shades are encoded in ./style.css.
+    These are 'cna' (grey), 'c0' (red), 'c1' ... 'c10' (green). The shade 'cna'
+    is used for items that are maked as 'not applicable'. The shades 'c1' to
+    'c9' form a gradient from red to lime-green to indicate 'levels of
+    completeness'. 'cna' is used for greying out a box for 'not applicable'
+    items, 'c0' is for items that are considered risky (or not yet started) and
+    'c10' for items that have completed successfully, or that are
+    'in good standing'.
+
+    These are the supported identifiers: %, %u, G, B, E, W, EN, WN.
+    The shading behavior for these is described below.
+
+    %:  Coloured percentage, where the number in front of the '%' sign is mapped
+        to a color for the cell ranging from red ('c0') to green ('c10').
+    %u: Uncoloured percentage, where no markup is applied and '%u' is replaced
+        with '%' in the output.
+    G:  This stands for 'Good' and results in a green cell.
+    B:  This stands for 'Bad' and results in a red cell.
+    E:  This stands for 'Errors' and the cell is colored with red if the number
+        in front of the indicator is larger than 0. Otherwise the cell is
+        colored with green.
+    W:  This stands for 'Warnings' and the cell is colored with yellow ('c6')
+        if the number in front of the indicator is larger than 0. Otherwise
+        the cell is colored with green.
+    EN: This stands for 'Errors Negative', which behaves the same as 'E' except
+        that the cell is colored red if the number in front of the indicator is
+        negative.
+    WN: This stands for 'Warnings Negative', which behaves the same as 'W'
+        except that the cell is colored yellow if the number in front of the
+        indicator is negative.
```

Comment:
Can these be folded into E and W respectively (E -> red if positive or negative number, green if 0).

Reply:
Unfortunately no, since I use the `EN` version for negative slack (and positive slack would be good in that case). We could fold `EN` and `WN` into `E` and `W` and introduce something like `TP` and `TN` (threshold positive, threshold negative) to achieve this functionality...

Reply:
Okay then, leave it the way it is. THanks for the clarification!